investigations/LOKI/loki_performance.md

```diff
@@ -0,0 +1,178 @@
+
+## Scope
+This document only covers performance benchmarks for the LOKI instrument based on the geometry described in the proceeding section. A general benchmarking exercise of the live data reduction in Mantid was performed [here](https://github.com/DMSC-Instrument-Data/documents/blob/master/investigations/Live%20Reduction/LiveReductionInvestigation.md) and this document may be referenced within. 
+Additionaly, this document only explores performance at a target rate of 10<sup>7</sup>events/s which corresponds to the predicted initial operational flux to 2MW. See [ess-hardware-requirements-report-v1.pdf](report.pdf) produced by Simon Heybrock for details.
+
+## LOKI Geometry
+The LOKI Geometry used for these tests are as specified in the engineering diagrams below. This design supersedes the initial BandGEM approach and will be utilizing a straw tube implementation. This decision to use straw tubes was fixed in early 2018. The instrument consists of 864 tubes (6048 straws) with an intrinsic resolution of 5mm. Straw lengths vary between 1.2m and 0.5m which gives a total pixel count of 1,245,880pixels.
+![geometry](loki_geometry.png)
+![mantid](loki_mantid.png)
+
+## Build-out Phases
+
+Currently "Day 1" operations for LOKI are set to commence November 2022. The initial instrument will contain the rear detector bank and 2 panels (one horizontal and one vertical) in the front and middle banks each. The resulting instrument will consist of ~473K pixels. However based on initial simulations, most of the detected neutron beam will be concentrated on the rear bank. The remaining banks are currently set to be added to the final instrument between 1-5 years beyond 2022. Hot commissioning and early science for this instrument are set for mid 2023 with full operations at the start of 2025.
+
+## Hardware 
+
+The hardware used for these tests is as follows:
+
+- 128GB RAM
+- Intel(R) Core(TM) i9-9920X CPU @ 3.50GHz
+- Target Linux Platform (OpenMP is a hard requirement)
+
+## Outcomes
+
+### Measurement Interpretation
+The ESS pulse rate for neutrons is expected to be **14Hz**. This will be the rate at which messaged will be produced in the Kafka system and therefore consumed by Mantid. Measurements are taken in this context. If Mantid is capable of processing events (writing events to in-memory data structures and performing basic reduction) at this rate at a target neutron flux, this indicates that Mantid can cope with the event rate.
+
+### Data ingest into Mantid
+
+Data consumption into Mantid using realistic SANS data generated using GEANT4 resulted in an average consumption rate of ~24Hz. This was performed at an event flux of ~1.4e<sup>7</sup> events per message. This clearly shows that Mantid can comfortably deal with the ingest of data at these high rates.
```

Comment:
10^8 was also mentioned. Was this measured, can anything be said about this? I don't we should worry if not as the Milestone did not mention it.

Reply:
Along these lines - how can we re-run all facets of the tests/measurements required for this report. It might be good to give links to all branches/repos or maybe there is an easier way? I don't think this should delay this report at all though.

Reply:
No I mentioned this in the scope. We limited the measurements to ~10e7 in line with the projected rates during Hot Commissioning/early science.

Reply:
@OwenArnold  The links to the various repos are within the document (e.g the kafka streamer). The only thing which is missing is Mantid, however all of the diagnostics are intrusive so we wouldn't have a release version of Mantid which will spit out benchmarks.